homeassistant/components/emulated_hue/config.py

```diff
@@ -0,0 +1,209 @@
+"""Support for local control of entities by emulating a Philips Hue bridge."""
+from collections.abc import Iterable
+import logging
+
+from homeassistant.const import CONF_ENTITIES, CONF_TYPE
+from homeassistant.core import State
+from homeassistant.helpers import storage
+
+TYPE_ALEXA = "alexa"
+TYPE_GOOGLE = "google_home"
+
+
+NUMBERS_FILE = "emulated_hue_ids.json"
+DATA_KEY = "emulated_hue.ids"
+DATA_VERSION = "1"
+SAVE_DELAY = 60
+
+CONF_ADVERTISE_IP = "advertise_ip"
+CONF_ADVERTISE_PORT = "advertise_port"
+CONF_ENTITY_HIDDEN = "hidden"
+CONF_ENTITY_NAME = "name"
+CONF_EXPOSE_BY_DEFAULT = "expose_by_default"
+CONF_EXPOSED_DOMAINS = "exposed_domains"
+CONF_HOST_IP = "host_ip"
+CONF_LIGHTS_ALL_DIMMABLE = "lights_all_dimmable"
+CONF_LISTEN_PORT = "listen_port"
+CONF_OFF_MAPS_TO_ON_DOMAINS = "off_maps_to_on_domains"
+CONF_UPNP_BIND_MULTICAST = "upnp_bind_multicast"
+
+
+DEFAULT_LIGHTS_ALL_DIMMABLE = False
+DEFAULT_LISTEN_PORT = 8300
+DEFAULT_UPNP_BIND_MULTICAST = True
+DEFAULT_OFF_MAPS_TO_ON_DOMAINS = ["script", "scene"]
+DEFAULT_EXPOSE_BY_DEFAULT = True
+DEFAULT_EXPOSED_DOMAINS = [
+    "switch",
+    "light",
+    "group",
+    "input_boolean",
+    "media_player",
+    "fan",
+]
+DEFAULT_TYPE = TYPE_GOOGLE
+
+ATTR_EMULATED_HUE_NAME = "emulated_hue_name"
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
```

Comment:
There is a lot of deprecated options that can probably go, but I didn't want to do too much in this PR